developer-docs-site/docs/move/book/functions.md

```diff
@@ -521,44 +521,58 @@ fun foo() { return () }
 
 ## Inline Functions
 
-Inline functions are functions which are expanded at caller side instead
-of compiled into Move bytecode. This allows to save gas and may lead
-to faster execution. For example, one can define an inline function
+Inline functions are functions whose bodies are expanded in-place at the caller location during compile time.
+Thus, inline functions do not appear in Move bytecode as a separate function: all calls to them are expanded away by the compiler.
+They allow to save gas and may lead to faster execution.
```

Comment:
The gas part is very dependent on circumstances and to whom:
1. If an inline function with nontrivial amount of code is used multiple times, the cost of deploying such a module would be much higher
2. The cost of loading such modules is also higher for the network but this is not charged yet
3. The save on an extra function call op is not very high to users I think

Reply:
```suggestion
In certain circumstances, they may lead to faster execution and save gas.
```

Would this address your points succinctly?

Reply:
This also fixes the grammar problem I tried to correct with a different suggestion.

Reply:
It makes sense as a summary. I wonder if we can provide the following guideline as to when to use inline functions:
1. For library functions that want to accept a lambda argument. Currently only inline functions can do this. Might change in Move v2
2. For functions that need to return references. This is common with global storage references (from borrow_global_mut or similar functions such as smart_table::borrow_mut)
3. For lightweight functions that are either very short (a couple of lines) or those not called repeated across the module. For very long inline functions, calling many times would bloat the bytecode of the module and lead to both higher gas cost when deploying and is generally wasteful storage-wise for the network.

Reply:
> For library functions that want to accept a lambda argument. Currently only inline functions can do this.

This phrasing is a bit ambiguous. Can you clarify the intent here, specifically what "this" at the end means?

-----

I like your idea of guidelines. I am wondering if a consolidated place for this would be a "move coding guidelines" page, perhaps merged along with the coding conventions (https://aptos.dev/move/book/coding-conventions)?

Another option is to add this a new section below here called "typical use cases for inline functions". I am leaning towards this.

-----

> For lightweight functions that are either very short (a couple of lines) 

With inline functions calling other inline functions , even with short functions, one can create exponential bloat. Perhaps we can also warn about transitive inline function calls?